sim/target-agnostic.mk

```diff
@@ -169,38 +186,27 @@ board_dir 	   := $(fpga_dir)/hdk/cl/developer_designs
 fpga_work_dir  := $(board_dir)/cl_$(name_tuple)
 fpga_build_dir := $(fpga_work_dir)/build
 verif_dir      := $(fpga_work_dir)/verif
-fpga_v         := $(fpga_work_dir)/design/cl_firesim_generated.sv
-ila_work_dir   := $(fpga_work_dir)/design/ila_files/
-fpga_vh        := $(fpga_work_dir)/design/cl_firesim_generated_defines.vh
-fpga_tcl_env   := $(fpga_work_dir)/design/cl_firesim_generated_env.tcl
 repo_state     := $(fpga_work_dir)/design/repo_state
 
+# Enumerates the subset of generated files that must be copied over for FPGA compilation
+fpga_delivery_files = $(addprefix $(fpga_work_dir)/design/$(BASE_FILE_NAME), \
+	.sv .defines.vh .env.tcl \
+	.ila_insert_inst.v .ila_insert_ports.v .ila_insert_wires.v .ila_insert_vivado.tcl)
+
 $(fpga_work_dir)/stamp: $(shell find $(board_dir)/cl_firesim -name '*')
 	mkdir -p $(@D)
 	cp -rf $(board_dir)/cl_firesim -T $(fpga_work_dir)
 	touch $@
 
-$(fpga_v): $(VERILOG) $(fpga_work_dir)/stamp
+$(repo_state): $(simulator_verilog) $(fpga_work_dir)/stamp
 	$(firesim_base_dir)/../scripts/repo_state_summary.sh > $(repo_state)
-	cp -f $< $@
-	sed -i "s/\$$random/64'b0/g" $@
-	sed -i "s/\(^ *\)fatal;\( *$$\)/\1fatal(0, \"\");\2/g" $@
 
-$(fpga_vh): $(VERILOG) $(fpga_work_dir)/stamp
+$(fpga_work_dir)/design/$(BASE_FILE_NAME)%: $(simulator_verilog) $(fpga_work_dir)/stamp
 	cp -f $(GENERATED_DIR)/$(@F) $@
 
-$(fpga_tcl_env): $(VERILOG) $(fpga_work_dir)/stamp
-	cp -f $(GENERATED_DIR)/$(@F) $@
-
-.PHONY: $(ila_work_dir)
-$(ila_work_dir): $(verilog) $(fpga_work_dir)/stamp
-	cp -f $(GENERATED_DIR)/firesim_ila_insert_* $(fpga_work_dir)/design/ila_files/
-	sed -i "s/\$$random/64'b0/g" $(fpga_work_dir)/design/ila_files/*
```

Comment:
This is now set with a verilog `define. And `SYNTHESIS hands the other case.

Reply:
This has the nice property of no longer needing to explicitly modify the output verilog for Vivado.